ruby/ql/lib/codeql/ruby/frameworks/GraphQL.qll

```diff
@@ -0,0 +1,270 @@
+private import codeql.ruby.AST
+private import codeql.ruby.Concepts
+private import codeql.ruby.controlflow.CfgNodes
+private import codeql.ruby.DataFlow
+private import codeql.ruby.dataflow.RemoteFlowSources
+private import codeql.ruby.ast.internal.Module
+
+
+private class GraphqlRelayClassicMutationAccess extends ConstantReadAccess {
+  //GraphQL::Schema::RelayClassicMutation
+  GraphqlRelayClassicMutationAccess() {
+    this.getName() = "RelayClassicMutation" and
+    this.getScopeExpr().(ConstantAccess).getName() = "Schema" and
+    this.getScopeExpr().(ConstantAccess).getScopeExpr().(ConstantAccess).getName() = "GraphQL"
```

Comment:
We can use API Graphs for these kinds of queries. We don't yet have documentation for the Ruby API Graphs library, but you can replace the above with
```ql
private import codeql.ruby.ApiGraphs

...

this = API::getTopLevelMember("GraphQL")
         .getMember("Schema")
         .getMember("RelayClassicMutation")
         .getAUse()
         .asExpr()
         .getExpr()
```

If you want you could then go futher and inline this class into its only use, in `GraphqlRelayClassicMutationClass`.